### PR TITLE
Remove deprecated Cloudflare AMP Cache and A4A support.

### PR DESCRIPTION
**Pull request by oliy**

### CODE REVIEW

File: validator/engine/validator-in-browser.js [hunk 1]
```diff
@@ -51,9 +51,7 @@ function getUrl(url) {
  * @return {boolean}
  */
 amp.validator.isAmpCacheUrl = function(url) {
-  return (
-    url.toLowerCase().indexOf('cdn.ampproject.org') !== -1 || // lgtm [js/incomplete-url-substring-sanitization]
-    url.toLowerCase().indexOf('amp.cloudflare.com') !== -1); // lgtm [js/incomplete-url-substring-sanitization]
+  return url.toLowerCase().indexOf('cdn.ampproject.org'); // lgtm [js/incomplete-url-substring-sanitization]
```
Comment (jridgewell) [inline validator/engine/validator-in-browser.js]: This should be `!== -1`

Comment (oliy) [reply validator/engine/validator-in-browser.js]: Oops.  Bad edit.  Fixed.

**Comment by Gregable** [comment]
Anything left here that needs further action for submitting?

**Comment by jridgewell** [comment]
Just a rebase

**Comment by oliy** [comment]
Rebasing!

File: build-system/test-configs/dep-check-config.js [hunk 2]
```diff
@@ -162,13 +162,6 @@ exports.rules = [
     allowlist: [
       // See todo note in ads/_a4a-config.js
       'ads/_a4a-config.js->' +
```
Comment (jridgewell) [inline build-system/test-configs/dep-check-config.js]: Ahh, need to delete this line, too.

Comment (oliy) [reply build-system/test-configs/dep-check-config.js]: Thanks!  Done.

**Comment by oliy** [comment]
I don't know why the visual change happened with the last change.  It shouldn't have affected things.  Is this PR ready for merging?

**Comment by jridgewell** [comment]
Seems like a flaky visual test. I manually approved, merging.

**Pull request merged**